src/components/form/fields/dateField.jsx

```diff
@@ -0,0 +1,170 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { DateTime, Info } from 'luxon'
+import styled from 'styled-components';
+
+import { Validator } from '../../../core/validation';
+import { useDebounce } from '../../../helpers/useDecounce';
+import { isValued } from '../../../helpers/utils';
+
+import { createField } from '../fieldCreator';
+import { FormGroup, Input, Select } from '../formControlsComponent';
+
+const inputRowGutter = 10;
+
+const InputRow = styled.div`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    gap: ${inputRowGutter}px;
+`;
+
+const InputCol = styled.div`
+    flex-basis: ${props => props.width}%;
+`;
+
+const DateField = ({ i18n, inputId, label, locale, onChange, path, required, showLabel, validation, value }) => {
+    const [day, setDay] = useState(isValued(value) ? value.raw.day : undefined)
+    const [month, setMonth] = useState(isValued(value) ? value.raw.month : undefined)
+    const [year, setYear] = useState(isValued(value) ? value.raw.year : undefined)
+
+    // debounce year value to delay value update when user is currently editing it
+    const debouncedYear = useDebounce(year, 1000)
+
+    const setDatePart = (setter, value) => {
+        if (Number.isNaN(Number(value))) return // only accept number value
+        setter(Number(value))
+    }
+
+    const handleDayChange = event => setDatePart(setDay, event.target.value)
+
+    const handleMonthChange = event => setDatePart(setMonth, event.target.value)
+
+    const handleYearChange = event => setDatePart(setYear, event.target.value)
+
+    useEffect(() => {
+        if (day && month && debouncedYear) {
+            onChange(() => ({
+                value: { raw: DateTime.fromObject({ year: debouncedYear, month, day }) },
+                isDirty: true,
+            }))
+        }
+    }, [debouncedYear, month, day]) // eslint-disable-line react-hooks/exhaustive-deps
+
+    const months = useMemo(() => Info.months("long", { locale }), [locale])
+    const daysInMonth = useMemo(() =>
+        Array.from({ length: DateTime.fromObject({ year: debouncedYear, month }).daysInMonth }, (_value, index) => index + 1),
+        [debouncedYear, month]
+    )
+
+    // reset day if current value is out of range
+    if (day && !daysInMonth.includes(day)) {
+        setDay(undefined)
+    }
+
+    const parts = useMemo(() =>
+        DateTime.now().setLocale(locale).toLocaleParts().map(part => part.type).filter(type => type !== 'literal'),
+        [locale]
+    )
+
+    const error = typeof validation === 'object' && 'error' in validation ? validation.error : null
+
+    return (
+        <FormGroup
+            inputId={inputId}
+            labelText={label}
+            error={error}
+            showLabel={showLabel}
```

Comment:
Notice an accessibility issue here.
`<FormGroup>` render an HTML `<label>` with a `for` attribute with `inputId` as value. But neither of the following inputs have this id in their attributs.
According to W3C, a label can refer to only one input, but we have 2 of them...
A valid solution would be to replace le HTML rendered by `<FormGroup>` with a `<fieldset>` and `<legend>`tag instead.